### PR TITLE
Character names can now have numbers

### PR DESCRIPTION
**Pull request by LittleBigKid2000**
For, you know, IPCs.

### CODE REVIEW

**Comment by tristan1333** [comment]
Oh god more things for IPC's

**Comment by Dinarzad** [comment]
Or conceivably Plasmamen. I've seen some use numerals in their names as it is an all that.

**Comment by LittleBigKid2000** [comment]
*Phew, just a speshul snowflek label, no yelling!*

**Comment by IcyV** [comment]
Should probably be restricted to IPCs only, if at all. Not really a fan of the implications of this if it's open to anyone.

**Comment by KasparoVy** [comment]
I'm sure in the highly advanced interstellar societies of 2700 it'd be perfectly normal to see a guy with a name like Dude 420 about.

**Comment by Dinarzad** [comment]
I still say it should also apply for Plasmamen, since their names are gas or element related half the time.

**Comment by Lady-Luck** [comment]
>Should probably be restricted to IPCs only, if at all. Not really a fan of the implications of this if it's open to anyone.

Snowflek code tho

File: code/modules/client/preference/preferences.dm [hunk 1]
```diff
@@ -1148,7 +1148,7 @@ var/global/list/special_role_times = list( //minimum age (in days) for accounts
 				if("name")
 					var/raw_name = input(user, "Choose your character's name:", "Character Preference") as text|null
 					if(!isnull(raw_name)) // Check to ensure that the user entered text (rather than cancel.)
-						var/new_name = reject_bad_name(raw_name)
+						var/new_name = reject_bad_name(raw_name, 2)
```
Comment (Fox-McCloud) [inline code/modules/client/preference/preferences.dm]: Why not just set this to `var/new_name = reject_bad_name(raw_name, 1)` and make no further changes.

**Comment by Rumiluntti** [comment]
I can finally be xxhungdaddy69xx

File: code/__HELPERS/text.dm [hunk 2]
```diff
@@ -115,7 +115,7 @@
 
 			// 0  .. 9
 			if(48 to 57)			//Numbers
-				if(!last_char_group)		continue	//suppress at start of string
+				if(!last_char_group && allow_numbers<2)		continue	//suppress at start of string
```
Comment (LittleBigKid2000) [inline code/__HELPERS/text.dm]: So that names can start with numbers

**Pull request merged**